txnsync/exchange.go

```diff
@@ -73,3 +75,18 @@ type timingParams struct {
 	AcceptedMsgSeq      []uint64 `codec:"a,allocbound=maxAcceptedMsgSeq"`
 	NextMsgMinDelay     uint64   `codec:"m"`
 }
+
+const (
+	noProposal byte = iota
+	transactionsForProposal
+	proposalByte
+	proposalFilter
+)
+
+type relayedProposal struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ProposalBytes   []byte        `codec:"pb,allocbound=maxEncodedTransactionGroupBytes"`
+	ProposalFilter  crypto.Digest `codec:"pf"`
+	ProposalMsgType byte          `codec:"mt"`
```

Comment:
add a comment describing when this would be used.
Generally speaking, when having the field of a data structure, we should avoid prefixing the data structure name from the fields. i.e. avoid `ProposalXXX`, since it's kind of weird to have this prefix on all the fields.

my suggestion is to use -
RawBytes, ExcludeProposal and Content

beside these, try to use a single character codec name as well as a different const for the byte array size limit.
we should probably pick a limit of 10K transactions for now.. so that's -
10K * 32 + sizeof(block header) ~= 350K